### PR TITLE
Stop requiring any sanitizers for libfuzzer

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Safe rust should largely not require AddrSan, so instead just build
without any sanitizers and let the fuzzer hunt for panics faster.

This speeds things up by about 2x on one of my targets, and appears to find new interesting targets without any changes.

### CODE REVIEW

**Comment by TheBlueMatt** [comment]
Note that the latest libfuzzer docs appear to suggest -sanitize=fuzzer for clang, which rustc, sadly, doesn't appear to support, so this may still break something, but I don't immediately see anything broken.

**Comment by Manishearth** [comment]
At least in the past libfuzzer wouldn't work if you didn't enable at least one sanitizer. This may have changed?

r? @fitzgen if you can verify stuff still works a

**Comment by TheBlueMatt** [comment]
It complains at startup, but appears to work without issue despite that, it may be that some things outside of simple fuzzing use-cases are broken. Eventually, someone with some knowledge of such infrastructure in rustc could add -Zsanitizer=fuzzer, but my brief glance didn't show a simple patch to add it.
```
WARNING: Failed to find function "__sanitizer_acquire_crash_state".
WARNING: Failed to find function "__sanitizer_print_stack_trace".
WARNING: Failed to find function "__sanitizer_set_death_callback".
```

File: src/options.rs [hunk 1]
```diff
@@ -78,8 +81,8 @@ pub struct BuildOptions {
     #[structopt(
         short = "s",
         long = "sanitizer",
-        possible_values(&["address", "leak", "memory", "thread"]),
-        default_value = "address",
+        possible_values(&["address", "leak", "memory", "thread", "none"]),
+        default_value = "none",
```
Comment (fitzgen) [inline src/options.rs]: I'm leaning towards leaving the default value as "address" for compatibility and for not silently moving folks who were relying on asan for their unsafe code away from this without them knowing.

@Manishearth do you have thoughts on this?

-------------------------

Either way we go with this we need a new test. If we keep the existing default, then let's add a test for using `-s none`. Otherwise, if we change the default to no sanitizer, then we need a test for enabling asan with `-s address`.

File: src/project.rs [hunk 2]
```diff
@@ -150,19 +150,23 @@ impl FuzzProject {
...
+                                     -Cllvm-args=-sanitizer-coverage-level=4 \
+                                     -Cllvm-args=-sanitizer-coverage-trace-compares \
+                                     -Cllvm-args=-sanitizer-coverage-inline-8bit-counters \
+                                     -Cllvm-args=-sanitizer-coverage-trace-geps \
+                                     -Cllvm-args=-sanitizer-coverage-prune-blocks=0 \
+                                     -Cllvm-args=-sanitizer-coverage-pc-table \
+                                     -Clink-dead-code"
+            .to_owned();
+        if let Sanitizer::None = build.sanitizer {
+        } else {
```
Comment (fitzgen) [inline src/project.rs]: This is a bit funky -- let's change it to either

* a `match`, or
* add a `is_none` method to `Sanitizer` so we can do `if !build.sanitizer.is_none() { ... }` here.

File: src/options.rs (hunk 1 above)
Comment (Manishearth) [reply src/options.rs]: Yes, would prefer to keep the default here.

Comment (TheBlueMatt) [reply src/options.rs]: Done, plus cloned the panic test without a sanitizer to have coverage for failure without any sanitizers.
